src/cmd-build

```diff
@@ -440,6 +440,12 @@ else
                  "--label=io.openshift.build.versions=machine-os=${buildid}"
                 )
     fi
+    previous_build=$(get_latest_build)
+    oci_tag=""
+    if [ -n "${previous_build}" ]; then
+      previous_builddir=$(get_build_dir "${previous_build}")
+      oci_tag+="--prior-build=oci-archive:${previous_builddir}/fedora-coreos-${previous_build}-ostree.${basearch}.ociarchive"
```

Comment:
Do we actually need the full OCI archive of the prior build for this, or just information about how the RPMs were binned? (Or just the RPM list?)

Having the pipeline fetch the previous build's OSTree is possible, but unfortunate.

Reply:
We only need the metadata, so indeed we could change things to fetch that.  However, there is no real sane way to fetch just the metadata from a .ociarchive - we'd need to fetch it from the registry aka quay.io/fedora/fedora-coreos:$branch

(BTW, this is what `rpm-ostree compose image` does in opinionated way; no s3 or ociarchives involved, just prior state in the registry)

Reply:
Running a `skopeo inspect docker://quay.io/fedora/fedora-coreos:[latest or parent]` should give the appropriate metadata (packing structure) for my usage. Agreed that we don't require the entire the `oci-archive`. 
In `cmd-build` we could invokes `skopeo inspect` and pass the `LayersData` field of the JSON to the `prior-build` of `container-encapsulate`

Reply:
Hmm, I'd rather not add a dependency on Internet access in `cosa build`. It increases risks of flakes and makes things less reproducible.

Could we just keep a copy of that data at build time in the build dir? Then we fetch it [when we `buildfetch` the parent build](https://github.com/coreos/fedora-coreos-pipeline/blob/6a8448ead37065bbc58925f0f88776c7ac478803/jobs/build.Jenkinsfile#L206-L214).

Reply:
Ok makes sense. At what step in cosa should `skopeo inspect ` be done or are you suggesting to do this separately?

Reply:
You can drop the JSON file into `tmp_builddir` after the bit where we `container-encapsulate`. Bikeshed name: `ocimeta.json` (by analogy to the existing `commitmeta.json`).

Reply:
thanks!

Reply:
We could also fetch the data at `cosa fetch` time from the registry.  I'm also fine with `ocimeta.json`.

Reply:
> You can drop the JSON file into `tmp_builddir` after the bit where we `container-encapsulate`. Bikeshed name: `ocimeta.json` (by analogy to the existing `commitmeta.json`).

I think the important choice over here is, whether we want to optimize the packing structure for the "user's" last build of FCOS versus if we want the packing structure to be optimized from the last stable build generally available to the users from the release browser.

The first option would make more sense since the layer deltas would be much lower for the user. But it could lead to the possibility where different users have different packing structures depending on their custom build. Does this matter?

Reply:
I think offhand...

- If `cosa buildfetch` is used, then we should default to using the last shipped stable build; this is the same use case as the production pipeline
- If `cosa buildfetch` wasn't used, then we should just use the previous local build

or so?

Reply:
Right, if it's in the build dir, then `cosa buildfetch` would fetch it. In the pipeline, we `buildfetch` *both* the previous build and the parent build, if available. If we found a parent build, we pass `--parent` to `cosa build`, so you'd know which metadata to use (parent vs previous build).

For RHCOS, we can't find out the parent build the same way (we don't currently publish a release index, and that wouldn't be enough anyway). Likely the simplest thing will be some code in cosa to make querying it easier so the pipeline knows which parent build to fetch.

Maybe to start though, let's just try to make this work well for FCOS. (Also, the RHCOS upgrade graph isn't as linear as FCOS, so an optimal packing is probably harder to achieve there anyway.)